src/nlp.jl

```diff
@@ -178,6 +178,18 @@ end
 
 const NonlinearConstraintRef = ConstraintRef{Model,NonlinearConstraintIndex}
 
+"""
+    all_nl_constraints(model::Model)
+
+Return a vector of all nonlinear constraint references in the model in the
+correct order.
```

Comment:
"the correct order" is ambiguous. It's better to say, "in the order in which they were added to the model".